Core/PaymentMethodHelper.php

```diff
@@ -203,4 +378,53 @@ public static function prepareCreditorName()
 
         return $sCreditorName ? $sCreditorName : $oShop->oxshops__oxcompany;
     }
+
+    /**
+     * Checks the user data if mandatory fields are set correctly for guaranteed invoice and saves them if needed
+     *
+     * @since 1.2.0
+     */
+    public static function checkPayStepUserInput()
+    {
+        $oUser = Registry::getSession()->getUser();
+
+        if (self::isDateOfBirthSet()) {
+            $oUser->oxuser__oxbirthdate = new Field(self::getDbDateOfBirth());
+        }
+
+        if (self::isPhoneValid()) {
+            $oUser->oxuser__oxfon = new Field(self::getPhone());
+        }
+
+        if (self::getSaveCheckoutFields() === '1') {
+            $oUser->save();
+        }
+
+        self::_validateUserInput();
+    }
+
+    /**
+     * Validates the user input and redirects to the payment step with an error if needed
+     *
+     * @since 1.2.0
+     */
+    private static function _validateUserInput()
```

Comment:
see comment above

Reply:
will need refactoring after everything is merged

Reply:
moved to payment methdod